### PR TITLE
fix(get_supported_scylla_base_versions): OSS upgrade version for 2024.1

### PR DESCRIPTION
**Pull request by fruch**
update the code, for the OSS upgrade path for 2024.1

### Testing

- [x] integration tests

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevent to this change (if needed)

### CODE REVIEW

**Review by vponomaryov — APPROVED** [review]
LGTM

File: utils/get_supported_scylla_base_versions.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 
 # We support to migrate from specific OSS version to enterprise
-supported_src_oss = {'2021.1': '4.3', '2022.1': '5.0', '2022.2': '5.1', '2023.1': '5.2', '2023.2': '5.3'}
+supported_src_oss = {'2021.1': '4.3', '2022.1': '5.0', '2022.2': '5.1', '2023.1': '5.2', '2024.1': '5.4'}
```
Comment (soyacz) [inline utils/get_supported_scylla_base_versions.py:17]: we no longer support 5.3->2023.2? while still from 4.3 to 2021.1?

Comment (fruch) [reply utils/get_supported_scylla_base_versions.py:17]: 2023.2 doesn't exists (and not going to be)
5.3 was canceled

Comment (soyacz) [reply utils/get_supported_scylla_base_versions.py:17]: true... :smile:

**Review by soyacz — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
integration tests issue are handled in #7043

**Pull request merged**